### PR TITLE
update `mkdirp` to v1.0.3

### LINKED ISSUE #2865: mkdirp < v1 deprecated
```bash
node-sass > mkdirp@0.5.3: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
```

mkdirp less than version 1 has been deprecated. I will try to submit a pull request later today

### PR DESCRIPTION
**Pull request by SampsonCrowley**
fixes #2865

### CODE REVIEW

**Comment by nschonni** [comment]
Looks like this only supports Node 8+, so it can't land on v4. Can you edit the PR target branch to `v5` and rebase if necessary?

**Comment by SampsonCrowley** [comment]
will do

**Comment by SampsonCrowley** [comment]
fixed

**Comment by SampsonCrowley** [comment]
@nschonni done

File: scripts/build.js [hunk 1]
```diff
@@ -24,28 +24,27 @@ function afterBuild(options) {
         : 'Release',
     'binding.node');
 
-  mkdir(path.dirname(install), function(err) {
-    if (err && err.code !== 'EEXIST') {
```
Comment (xzyfer) [inline scripts/build.js]: It looks like this behaviour has been changed now

**Comment by PolyPik** [comment]
I don't think we need this anymore due to [`fs.mkdir`](https://nodejs.org/docs/latest-v10.x/api/fs.html#fs_fs_mkdir_path_options_callback) allowing a recursive option that does the same thing.

**Comment by xzyfer** [comment]
@PolyPik the recursive options only landed in v10.12.0 so we can't rely on it yet.

**Comment by xzyfer** [comment]
I've rebased this branch on the latest v5

**Comment by PolyPik** [comment]
We can merge this now.

**Comment by IdanAdar** [comment]
@xzyfer Can you please also update `gulp-sass-lint` which you maintain with the same fix?

**Pull request merged**